lib/src/config/config.rs

```diff
@@ -119,7 +119,11 @@ impl Config {
                 return Err(self.bad_type(name, val, "an unsigned integer"));
             }
 
-            self.port = port as usize;
+            if port > (u16::max_value() as i64) {
```

Comment:
@mgattozzi If you do `let port: u16`, you'll truncate the value. This means it can never exceed `u16::max`, and values above `u16::max` will be accept unexpectedly.